flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -204,12 +204,19 @@ public void run() {
 					// we can close this consumer thread once we've reached the end of the subscribed shard
 					break;
 				} else {
+					long runLoopTimeSeconds = 0;
 					if (fetchIntervalMillis != 0) {
 							long elapsedTimeNanos = System.nanoTime() - lastTimeNanos;
 							long sleepTimeMillis = fetchIntervalMillis - (elapsedTimeNanos / 1_000_000);
+
+							// Time the sleep
+							long sleepStartTimeMillis = System.nanoTime();
```

Comment:
Get rid of this call to System.nanoTime() and instead just save the one from above which is used to calculate `elapsedTimeNanos`.  These two values are virtually identical.